python_on_whales/components/pod/cli_wrapper.py

```diff
@@ -0,0 +1,778 @@
+
+    def unpause(self) -> None:
+        """Unpause the pod
+
+        See the [`docker.pod.unpause`](../sub-commands/pod.md#unpause) command for
+        information about the arguments.
+        """
+        return PodCLI(self.client_config).unpause(self)
+
+    def restart(self) -> None:
+        """Restart this pod.
+
+        See the [`docker.pod.restart`](../sub-commands/pod.md#restart) command for
+        information about the arguments.
+        """
+        return PodCLI(self.client_config).restart(self)
+
+    def remove(self, *, force: bool = False, time: Optional[int] = None) -> None:
+        """Remove this pod.
+
+        See the [`docker.pod.remove`](../sub-commands/pod.md#remove) command for
+        information about the arguments.
+        """
+        PodCLI(self.client_config).remove(self, force=force, time=time)
+
+    def start(self) -> Union[None, str, Iterable[Tuple[str, bytes]]]:
+        """Starts this pod.
+
+        See the [`docker.pod.start`](../sub-commands/pod.md#start) command for
+        information about the arguments.
+        """
+        return PodCLI(self.client_config).start(self)
+
+    def stop(self, *, time: Optional[int] = None) -> None:
+        """Stops this pod.
+
+        See the [`docker.pod.stop`](../sub-commands/pod.md#stop) command for
+        information about the arguments.
+        """
+        return PodCLI(self.client_config).stop(self, time=time)
+
+
+ValidPod = Union[str, Pod]
+
+
+class PodCLI(DockerCLICaller):
+    def __init__(self, client_config: ClientConfig):
+        super().__init__(client_config)
+
+    def create(
+        self,
+        name: Optional[str] = None,
+        *,
+        add_hosts: List[Tuple[str, str]] = [],
+        cgroup_parent: Optional[str] = None,
+        cpus: Optional[float] = None,
+        cpuset_cpus: Optional[List[int]] = None,
+        devices: List[str] = [],
+        device_read_bps: List[str] = [],
+        dns: List[str] = [],
+        dns_options: List[str] = [],
+        dns_search: List[str] = [],
+        exit_policy: Optional[str] = None,
+        gidmaps: List[Tuple[int, int, int]] = [],
+        hostname: Optional[str] = None,
+        infra: Optional[bool] = None,
+        infra_command: Optional[str] = None,
+        infra_conmon_pidfile: Optional[ValidPath] = None,
+        infra_image: Optional[
+            python_on_whales.components.image.cli_wrapper.ValidImage
+        ] = None,
+        infra_name: Optional[str] = None,
+        ip: Optional[str] = None,
+        ip6: Optional[str] = None,
+        labels: Dict[str, str] = {},
+        label_files: List[ValidPath] = [],
+        mac_address: Optional[str] = None,
+        memory: Union[int, str, None] = None,
+        networks: List[
+            python_on_whales.components.network.cli_wrapper.ValidNetwork
+        ] = [],
+        network_aliases: List[str] = [],
+        no_hosts: bool = False,
+        pid: Optional[str] = None,
```

Comment:
Shouldn't this be an int?

Reply:
There's an inconsistency in the CLI here... The following are used for controlling namespaces: `--cgroupns` (cgroup), `--network` (net), `--pid` (pid), `--userns` (user).
```
$podman pod create --help | grep -- --pid
      --pid string                    PID namespace to use
```
This will be clearer when I've added the method docstring :)